roq-data/deployment/src/main/java/io/quarkiverse/roq/data/deployment/RoqDataConfig.java

```diff
@@ -19,10 +20,17 @@ public interface RoqDataConfig {
     @WithDefault(DEFAULT_LOCATION)
     String dir();
 
+    /**
+     * Whether to enforce the use of a bean for each data file.
+     * <br>
+     * With this option enabled, when a record is annotated with {@link DataMapping}, a bean will be created and populated
+     * with the data from the file.
+     */
+    @WithDefault("false")
+    boolean enforceBean();
+
     static boolean isEqual(RoqDataConfig q1, RoqDataConfig q2) {
-        if (!Objects.equals(q1.dir(), q2.dir())) {
-            return false;
-        }
-        return true;
```

Comment:
Don't you also need to check enforceBean in the isEqual?  Typically we compare all the props.